# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcCallWrapper.h
# Commit: af4c24767e48
# Full Hash: af4c24767e48def595991abacc5719eb77e1b27f
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Make mediapipeline_unittest pass and remove TaskQueueWrapper::MainAsCurrent. r=ng,bwc
#   
#   This patch makes mediapipeline_unittest pass an event target to the Call wrapper
#   but keeps using main thread for that event target, so that it can keep the test
#   cases simpler by allowing sybnc calls into the conduits.
# ==============================================================================

diff -r d621668762f4 -r af4c24767e48 dom/media/webrtc/libwebrtcglue/WebrtcCallWrapper.h
--- a/dom/media/webrtc/libwebrtcglue/WebrtcCallWrapper.h	Wed Apr 07 15:01:03 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/WebrtcCallWrapper.h	Fri Apr 09 10:26:12 2021 +0200
@@ -34,7 +34,7 @@
       const dom::RTCStatsTimestampMaker& aTimestampMaker,
       UniquePtr<media::ShutdownBlockingTicket> aShutdownTicket,
       RefPtr<AbstractThread> aCallThread,
-      rtc::scoped_refptr<webrtc::SharedModuleThread> aModuleThread,
+      std::function<webrtc::SharedModuleThread*()> aModuleThreadGetter,
       const webrtc::AudioState::Config& aAudioStateConfig,
       webrtc::AudioDecoderFactory* aAudioDecoderFactory,
       webrtc::WebRtcKeyValueConfig* aTrials);